pkg/route/route.go

```diff
@@ -0,0 +1,44 @@
+// Package route contains utilities for handling OpenShift route objects.
+package route
+
+import (
+	"fmt"
+	"net/url"
+
+	routev1 "github.com/openshift/api/route/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// IngressURI calculates an admitted ingress URI.
+// If 'host' is set, only the ingress for that host is considered.
+// If 'host' is unset, the first admitted ingress is used.
```

Comment:
Instead of "set" and "unset", maybe better to say "nonempty" and "empty."

Reply:
> Instead of "set" and "unset", maybe better to say "nonempty" and "empty."

Done in 08e24b164 -> c125aa6f7.